flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/CreateTableASOperation.java

```diff
@@ -26,25 +26,32 @@
 import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.function.Supplier;
 
 /** Operation to describe a CREATE TABLE AS statement. */
 @Internal
 public class CreateTableASOperation implements CreateOperation {
 
     private final CreateTableOperation createTableOperation;
-    private final SinkModifyOperation insertOperation;
+    private final Supplier<SinkModifyOperation> insertOperationFactory;
+
+    private SinkModifyOperation insertOperation;
```

Comment:
no mutability in operations, pass the resolved table to `getInsertOperation(...)` instead.

Reply:
I think I can't, check `asSummaryString`.

Reply:
I've modified a bit the structure of `CreateTableASOperation` to make the summary string senseful (and similar to `SinkModifyOperation`) and remove that internal mutability. check now